### PR TITLE
New Filetype: PL/SQL

### PR DESCRIPTION
**Pull request by somini**
Basic PL/SQL support. Filters away the arguments.

Includes support for nudist-mode.

PL/SQL is Oracle-specific.

### CODE REVIEW

File: autoload/ctrlp/funky/ft/sql.vim [hunk 1]
```diff
@@ -0,0 +1,21 @@
+" Language: SQL
+" Author: somini
+" License: The MIT License
+
+let s:pattern = '\V\c\<function\>\W\+\w\+(\_.\{-})'
```
Comment (tacahiroy) [inline autoload/ctrlp/funky/ft/sql.vim]: I tested and Funky showed nothing.
You need to escape parentheses to group the pattern with no very magic (\V).

```sql
CREATE OR REPLACE FUNCTION RIVUS.PRODUCT
(
	P_NUM1 IN NUMBER,
	P_NUM2 IN NUMBER
)
RETURN NUMBER
IS
	vSum	NUMBER(2) DEFAULT  0;
--	vSum	NUMBER(2) :=  0;
BEGIN
	FOR i IN 1..P_NUM2
	LOOP
		vSum := vSum + P_NUM1;
	END LOOP;
	RETURN vSum ;
END;
```

**Comment by somini** [comment]
You are correct, you can have functions namespaced by schema and functions without any parenthesis.
https://docs.oracle.com/cd/B28359_01/appdev.111/b28370/create_function.htm
Here's a nice test case.
```sql
create function tst()
create or replace function schema.test()
create or replace function schema.test_args(multi number,
						line varchar2,
						arguments number)
create or replace function schema.test_noparens
create or replace function schema.test_noparens_args(arguments number)
```

I've pushed my fixes to the regexes.

**Comment by tacahiroy** [comment]
This is perfect. Thanks!

**Comment by tacahiroy** [comment]
I've just noticed that Vim supports PL/SQL natively for extensions `.pls` or `.plsql`, so will copy sql.vim to plsql.vim.

**Comment by somini** [comment]
I've been using vim for ages and didn't notice that. Please do.

**Comment by somini** [comment]
Oh, I noticed another error in the regex, `\V\.´ means any character, so it matches comments with "function" anywhere.
Please change the "s:pattern" variable to `\V\c\<function\>\s\+\(\k\|.\)\+\((\_.\{-})\)\?`, I'm not sure if it's worth it to open a PR to delete a single character.

**Pull request merged**